### PR TITLE
Dns delegation clean azure

### PR DESCRIPTION
**Pull request by whites11**
See: https://github.com/giantswarm/giantswarm/issues/7981

This PR extends the ci-cleaner functionality with the goal of deleting NS records from the azure.gigantic.io zone when the relative api.* record does not resolve anymore.

### CODE REVIEW

File: pkg/cleaner/azure/delegatedns.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+			c.logger.LogCtx(ctx, "level", "error", "message", "skipping")
+			lastError = err
+			continue
+		}
+
+		if del {
+			c.logger.LogCtx(ctx, "level", "info", "message", fmt.Sprintf("DNS record %s has to be deleted", *record.Name))
+			err := c.deleteRecord(ctx, record)
+			if err != nil {
+				c.logger.LogCtx(ctx, "level", "error", "message", fmt.Sprintf("failed to check DNS record %q", *record.Name), "stack", fmt.Sprintf("%#v", microerror.Mask(err)))
```
Comment (fiunchinho) [inline pkg/cleaner/azure/delegatedns.go]:
```suggestion
				c.logger.LogCtx(ctx, "level", "error", "message", fmt.Sprintf("failed to delete DNS record %q", *record.Name), "stack", fmt.Sprintf("%#v", microerror.Mask(err)))
```

File: pkg/cleaner/azure/delegatedns.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+func (c Cleaner) cleanDelegateDNSRecords(ctx context.Context) error {
+	var lastError error
+
+	recordsIter, err := c.dnsRecordSetsClient.ListAllByDNSZoneComplete(ctx, resourceGroup, zoneName, nil, "")
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	deadLine := time.Now().Add(-gracePeriod).UTC()
```
Comment (fiunchinho) [inline pkg/cleaner/azure/delegatedns.go]: Where is `gracePeriod` coming from?

File: pkg/cleaner/azure/delegatedns.go (hunk 1 above)
Comment (whites11) [reply pkg/cleaner/azure/delegatedns.go]: thx

File: pkg/cleaner/azure/delegatedns.go (hunk 2 above)
Comment (whites11) [reply pkg/cleaner/azure/delegatedns.go]: It's defined in another file, resourcegroup.go

**Review by fiunchinho — APPROVED** [review]

**Comment by corest** [comment]
Any blockers for merging?

**Comment by whites11** [comment]
> Any blockers for merging?

Nope, just been blocked by other things.

**Pull request merged**